hoover/search/management/commands/doctor.py

```diff
@@ -0,0 +1,117 @@
+from pathlib import Path
+import re
+import sys
+import urllib.request
+import urllib.error
+import json
+from django.core.management.base import BaseCommand
+from django.conf import settings
+from django.db import connections
+from django.db.utils import OperationalError
+
+ELASTICSEARCH_MIN_VERSION = (2, 0, 0)
+ELASTICSEARCH_MAX_VERSION = (2, 4, 4)
+
+
+def http_get_content(link):
+    try:
+        with urllib.request.urlopen(link) as content:
+            return content.read()
+    except (urllib.error.HTTPError, urllib.error.URLError):
+        return None
+
+
+class Command(BaseCommand):
+    help = "Sanity check for search. Run with no arguments."
+
+    def handle(self, *args, **options):
+        checkers = [
+            ('python', self.check_python, False),
+            ('database', self.check_database, False),
+            ('elasticsearch', self.check_es, False),
+            ('events_dir', self.check_events_dir, settings.HOOVER_EVENTS_DIR is None)
+        ]
+
+        have_errors = False
+        for name, check_fun, skip in checkers:
+            if skip:
+                self.print_message("Skipping the check for " + name + ".")
+            else:
+                self.print_message("Checking " + name + ".")
+                result = check_fun()
+                if result:
+                    self.print_success(' ' * 9 + name + " ok.")
+                else:
+                    have_errors = True
+                    self.print_error(name + " failed the check.")
+            self.print_message('')
+
+        if have_errors:
+            self.print_error("The setup has failed some checks.")
+            self.print_error("For more information please see")
+            self.print_error("https://github.com/hoover/search/blob/master/Readme.md")
+            sys.exit(1)
+        else:
+            self.print_success("All checks have passed.")
+
+    def check_python(self):
+        if sys.version_info[0] != 3 or sys.version_info[1] < 5:
```

Comment:
I'm a bit afraid of a python >= 4.
Who knows what code archaeologist will want to install the stack?